compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -792,8 +792,11 @@ impl<'hir> LoweringContext<'_, 'hir> {
         // debuggers and debugger extensions expect it to be called `__awaitee`. They use
         // this name to identify what is being awaited by a suspended async functions.
         let awaitee_ident = Ident::with_dummy_span(sym::__awaitee);
-        let (awaitee_pat, awaitee_pat_hid) =
-            self.pat_ident_binding_mode(span, awaitee_ident, hir::BindingAnnotation::MUT);
+        let (awaitee_pat, awaitee_pat_hid) = self.pat_ident_binding_mode(
+            gen_future_span,
```

Comment:
we use the whole expression span here rather than the span of the `.await`, when pointing to the local that we use to store the `IntoFuture::into_future` result. That's because the user probably cares more about the whole async call, not just the fact that the future was awaited.

This matters a lot more when pointing out recursive calls in #117703, but it does make the span a bit better here too.